src/debug/HyprCtl.cpp

```diff
@@ -1347,13 +1347,20 @@ std::string getReply(std::string request) {
     auto format = HyprCtl::FORMAT_NORMAL;
 
     // process flags for non-batch requests
-    if (!request.contains("[[BATCH]]") && request.contains("/")) {
-        long unsigned int sepIndex = 0;
+    if (!request.starts_with("[[BATCH]]") && request.contains("/")) {
+        uint8_t sepIndex = 0;
```

Comment:
why?

Reply:
Unfortunately my week old cpp experience cannot explain (yet) why not. It seemed reasonable, since there won't be that many flags.

Reply:
@dranull `long unsigned int` == `size_t` or `uint64_t`, better use `size_t`,
`uint8_t` has max value = 255 and we are limited to 255 chars string. Based on the logic of the code, you are getting an integer overflow. `sepIndex >= UINT8_MAX`. You have broken the logic of the code. The code parses the string and looks for '/', and removes everything before /. This way you have limited string to 255 characters.

Reply:
@h0tc0d3 I see, though it doesn't overflow (because of the check), but indeed it will cut the string to 255, and `if (sepIndex < request.size())` will be true then. Every request I tested was way under 255, and I was focusing on the length of the flags to limit them further, so I didn't notice. My bad, sorry.

Reply:
@dranull Don't create empty commits when you create a commit and then delete it.

Use interactive rebase instead. `git rebase -i COMMIT_HASH^`
Replace `pick` with `drop` or `edit`. 
https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History 
After changes: `git push --force`

Reply:
I know how that works, but I don't see the point in this case, it's a new commit not a deleted one. The commits are squashed on merge anyway, so you'll have only one commit in the repo, from which you can't tell what the individual commits did. But in the pull request at least you'll be able to see the review process. But if the boss wishes, I'll do that way.